x-pack/legacy/plugins/maps/public/layers/util/es_agg_utils.ts

```diff
@@ -43,6 +44,13 @@ export function extractPropertiesFromBucket(bucket: any, ignoreKeys: string[] =
       properties[key] = bucket[key].value;
     } else if (_.has(bucket[key], 'buckets')) {
       properties[key] = _.get(bucket[key], 'buckets[0].key');
+      const topBucketCount = bucket[key].buckets[0].doc_count;
```

Comment:
this can be empty, when the top-term field is sparsely populated.

e.g.

```
        {
          "key": {
            "gridSplit": "14/4283/6311"
          },
          "doc_count": 1,
          "terms_of_name.keyword": {
            "doc_count_error_upper_bound": 0,
            "sum_other_doc_count": 0,
            "buckets": []
          },
          "gridCentroid": {
            "location": {
              "lat": 38.14009466674179,
              "lon": -85.88920757174492
            },
            "count": 1
          }
        },
```

Unchecked, the error bubbles up in the UX

![image](https://user-images.githubusercontent.com/1833023/76022448-2d67d780-5ef5-11ea-9566-0868a6d58154.png)



Since we know _exactly_ what response we want and what edge-cases we want to test for and handle, I think using `_.get` is not necessary.

Reply:
Added test case to handle empty buckets array.